### DISCUSSION TITLE
v9 ".start" error

### DISCUSSION THREAD

**Discussion by carmandale** [pmndrs/use-gesture#291] [Q&A] [answered]
I am getting this error
react-spring: Directly calling start instead of using the api object is deprecated in v9 (use ".start" instead)

I am using the example from here: 
https://use-gesture.netlify.app/docs/options/

`function Box() {
  const { viewport } = useThree()
  const { width, height, factor } = viewport
  const [spring, setSpring] = useSpring(() => ({ position: [0, 0, 0], scale: [1, 1, 1] }))

  const bind = useDrag(({ offset: [x, y] }) => setSpring({ position: [x, y, 0] }), {
    // bounds are expressed in canvas coordinates!
    bounds: { left: -width / 2, right: width / 2, top: -height / 2, bottom: height / 2 },
    rubberband: true,
    transform: ([x, y]) => [x / factor, -y / factor],
  })

  return (
    <a3f.mesh {...bind()} {...spring}>
      <boxBufferGeometry args={[1, 1, 1]} />
      <meshStandardMaterial color="orange" />
    </a3f.mesh>
  )
}
function Transform() {
  return (
    <Canvas>
      <ambientLight intensity={0.5} />
      <spotLight position={[10, 10, 10]} angle={0.15} penumbra={1} />
      <Box />
    </Canvas>
  )
}`

I am not sure where or how to use the .start. Can someone clarify? I pulled this from the documentation.

thanks,
Dale

**Comment by dbismut** [comment, marked as answer]
I forgot to update the example but use setSpring.start.

**Reply by carmandale** [reply]
Thank you. That removes the error. It works on desktop and on iOS mobile just as expected. 
But on the chrome, inspector - mobile simulator, where you can make it "mobile" the drag is greatly reduced, basically unusable, and I get the same on Android mobile too.

any ideas?

**Reply by dbismut** [reply]
touch-action: none on the canvas.

**Reply by carmandale** [reply]
Hallelujah! that did it.
